lib/evmmax/evmmax.cpp

```diff
@@ -0,0 +1,114 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2023 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <evmmax/evmmax.hpp>
+
+using namespace intx;
+
+namespace evmmax
+{
+namespace
+{
+inline constexpr uint64_t mul_inv64(uint64_t base) noexcept
```

Comment:
Can you explain this too? what does it calculate and what's the forumla?

Also, why is used only with 1 word of modulus.

Reply:
I don't know exactly. I copied it from somewhere. But this is some Montgomery multiplication magic. I'm guessing this is `R⁻¹ mod N` but only most significant bits are needed.

Reply:
Ok, this is used to compute "Integer N′ in [0, B − 1] such that NN′ ≡ −1 (mod B)" where B is `2^64`. This loop looks like exponentiation with exponent being `−1 (mod B)` i.e. `2^64-1`.